src/dataflow-types/src/client/replicated.rs

```diff
@@ -297,12 +301,15 @@ where
                 use futures::StreamExt;
                 while let Some((replica_id, message)) = stream.next().await {
                     match message {
-                        Ok(ComputeResponse::PeekResponse(uuid, response)) => {
+                        Ok(ComputeResponse::PeekResponse(uuid, response, otel_ctx)) => {
                             // If this is the first response, forward it; otherwise do not.
                             // TODO: we could collect the other responses to assert equivalence?
                             // Trades resources (memory) for reassurances; idk which is best.
+                            // TODO(guswynn): consider if the `otel_ctx` is correct here
```

Comment:
Seems fine! If you store `otel_ctx` in `self.peeks` then you could alternatively pull it out of `self.peeks`, but it'll be identical!

Reply:
actually we don't want to pull it out, cause the `PeekResponse` otel_ctx is different, its the one the compute worker made!

Reply:
Ohh, I see, it has a different parent span?